### PR TITLE
Graphics update!

### PR DESCRIPTION
**Pull request by kirazy**
I made these masks for Krastorio... but I thought rather than try to make a mod to reskin Miniloaders it would be so much easier to just... update Miniloaders.

Specific changes:
- New graphics and masks for loader, inserter, item, and technology.
- Added color support for the Ultimate Belt reskin.
- Went through and calibrated the tints for all the supported mods.
- Added a function to obtain tints from hex color codes.

The core miniloaders:
image

Ultimate belts, as an example, but all the supported mods have updated color matching:
image

As an aside, I think you can probably drop support for Xander's mod and the old Krastorio mod. In Xander's case, there's no more expedited belt that I could find. And the old Krastorio mod is broken by a recent vanilla update (and is deprecated so will never work again).

### CODE REVIEW

**Comment by kirazy** [comment]
Oh, forgot to include pictures of in-game icons/tech icons:
image
image

File: prototypes/entities.lua [hunk 1]
```diff
@@ -38,91 +38,161 @@ local function create_loaders(prefix, base_underground_name, tint)
...
-          height = 43,
-          y = 43,
-          hr_version = {
-            filename = "__miniloader__/graphics/entity/hr-template.png",
+          filename = "__miniloader__/graphics/entity/miniloader-structure-base.png",				
+          width    = 96,
+          height   = 96,
+          y        = 0,
+          hr_version = 
+          {
```
Comment (mspielberg) [inline prototypes/entities.lua]: To keep style consistent, could you change these to not vertically align the `=`, and put `{` on the same line as the `=`, like you do with front_patch and back_patch?

File: prototypes/templates.lua [hunk 2]
```diff
@@ -1,40 +1,68 @@
+-- Converts hex code values to rgb values, alpha is an optional parameter
+local function tint_from_hex(hex, alpha)
```
Comment (mspielberg) [inline prototypes/templates.lua]: Could you use [util.color](https://github.com/wube/factorio-data/blob/master/core/lualib/util.lua#L65-L78)?

File: data.lua [hunk 3]
```diff
@@ -21,7 +21,7 @@ end
 if settings.startup["miniloader-enable-chute"].value then
   local standard_enabled = settings.startup["miniloader-enable-standard"].value
   settings.startup["miniloader-enable-standard"].value = true
-  create_miniloader("chute-", "", {"logistics"}, {r=0.5,g=0.5,b=0.5}, "underground-belt")
+  create_miniloader("chute-", "", {"logistics"}, {r = 0.5, g = 0.5, b = 0.5, a = 0.82}, "underground-belt")
```
Comment (mspielberg) [inline data.lua]: Could you use [util.color](https://github.com/wube/factorio-data/blob/master/core/lualib/util.lua#L65-L78) for consistency with below?

**Review by mspielberg — CHANGES_REQUESTED** [review]
These look amazing! Thanks so much for putting in the work on all of this. I wasn't sure about the future status of Krastorio Legacy, but if it's no longer being updated it makes sense to drop support. I have a few minor stylistic comments. I can take care of them if you'd prefer, and I've love to get this merged and released to the world!

**Comment by kirazy** [comment]
I'm happy to make the changes myself. Also thanks for bringing util.color to my attention, now I have to go make some changes after this in another of my mods...

**Comment by kirazy** [comment]
Finished make the changes you asked for.

**Comment by bitwiseshiftleft** [comment]
Thanks for all your work on this!

I do have a concern about the new graphics, though.  In my opinion the filtered and unfiltered miniloaders look too similar, so that it's hard to tell them apart at a glance either in inventory or once placed.  (Edit: This was a problem in the previous graphics on the red loaders, but not as much on other colors.)  Also, vertically oriented miniloaders and underneathies are very similar, despite doing quite different things.

So maybe next time you change the graphics, could you consider making them more visually distinct?  Like maybe they could have some kind of controller on top, that would glow or otherwise be a different color on filtered miniloaders, instead of just a pink tint to the metal?

**Pull request merged**